decidim-core/app/presenters/decidim/log/base_presenter.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Log
+    # This class holds the logic to present a resource for any activity log.
+    # It is supposed to be a base class for all other log renderers, as it defines
+    # some helpful methods that later presenters can use.
+    #
+    # Most presenters that inherit fromt his class will only need to overwrite
+    # the `action_string` method, which defines what I18n key will be used for
+    # each action.
+    #
+    # See `Decidim::ActionLog#render_log` for more info on the log types and
+    # presenters.
+    #
+    # Usage should be automatic and you shouldn't need to call this class
+    # directly, but here's an example:
+    #
+    #    action_log = Decidim::ActionLog.last
+    #    view_helpers # => this comes from the views
+    #    BasePresenter.new(action_log, view_helpers).present
+    class BasePresenter
+      # Public: Initializes the presenter.
+      #
+      # action_log - An instance of Decidim::ActionLog.last
+      # view_helpers - An object holding the view helpers at the render time.
+      #   Most probably should come automatically from the views.
+      def initialize(action_log, view_helpers)
+        @action_log = action_log
+        @view_helpers = view_helpers
+      end
+
+      # Public: Renders the resource associated to the given `action_log`.
+      #
+      # action_log - An instance of Decidim::ActionLog.last
+      # view_helpers - An object holding the view helpers at the render time.
+      #   Most probably should come automatically from the views.
+      #
+      # Returns an HTML-safe String.
+      def present
+        h.content_tag(:li) do
+          I18n.t(
+            action_string,
+            i18n_params
+          ).html_safe +
+            " (#{h.localize(action_log.created_at, format: :decidim_short)})"
+        end
+      end
+
+      private
+
+      attr_reader :action_log, :view_helpers
+      alias h view_helpers
+
+      delegate :action, to: :action_log
+
+      def user
```

Comment:
Maybe it's easier to delegate all these methods?

Reply:
I tried delegating, but then I get a lot of cached queries in the logs, so I thought it was better if we memoized this in the Ruby level.

I'll add some docs explaining the reasoning.

Reply:
Done